common/src/main/java/com/wynntils/models/labels/LabelModel.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.labels;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Model;
+import com.wynntils.mc.event.SetEntityDataEvent;
+import com.wynntils.utils.mc.McUtils;
+import java.util.List;
+import java.util.Optional;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.syncher.SynchedEntityData;
+import net.minecraft.world.entity.Entity;
+import net.minecraftforge.eventbus.api.EventPriority;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class LabelModel extends Model {
```

Comment:
Label model? I think models should model some kind of concept present in the game. So "ProfessionModel" sounds perfectly fine. But here I think you are more trying to generalize a way to handle Minecraft data. I think you should be able to do the onEntitySetData where you need it. And if you really want to have this somewhere independent, it is more a matter of a "LabelHandler". But that would also kind of imply that there are (or are planned to soon be) more than one usage of the LabelEvent.

Reply:
There will be more use case for it, area dps iirc.

Reply:
Ok, if so, I think it will make more sense being a LabelHandler.